nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -209,54 +209,90 @@ trait NirGenExpr { self: NirGenPhase =>
 
     def genMatch(m: Match): Val = {
       val Match(scrutp, allcaseps) = m
+      type Case = (Local, Val, Tree, nir.Position)
 
       // Extract switch cases and assign unique names to them.
-      val caseps: Seq[(Local, Val, Tree)] = allcaseps.flatMap {
+      val caseps: Seq[Case] = allcaseps.flatMap {
         case CaseDef(Ident(nme.WILDCARD), _, _) =>
           Seq()
-        case CaseDef(pat, guard, body) =>
-          assert(guard.isEmpty)
+        case cd @ CaseDef(pat, guard, body) =>
+          assert(guard.isEmpty, "CaseDef guard was not empty")
           val vals: Seq[Val] = pat match {
             case lit: Literal =>
-              List(genLiteralValue(lit))
+              List(genExpr(lit))
             case Alternative(alts) =>
               alts.map {
-                case lit: Literal => genLiteralValue(lit)
+                case lit: Literal => genExpr(lit)
```

Comment:
Why can't we use `genLiteralValue` here anymore?

Reply:
We can actually, I've fixed it